### ISSUE TITLE
@turf.difference returns a GeometryCollection

### ISSUE THREAD

**Issue by deregtd** [Turfjs/turf#1779]
I haven't exported the exact set of data that causes the situation yet, but I'm using turf/difference to clip regions out of geo data, and I'm running into a bunch of circumstances where, after running turf/difference, it actually returns a GeometryCollection, not a polygon/multipolygon.  In the instances I'm seeing, the returned geometrycollection is a set of one Polygon and one LineString.  I'm not sure why it's doing this (I haven't plotted the results on a map yet to see what's up), but in any event, the typescript typings are wrong and I'm having to cast things around as any to make types work out, so if difference can return a GeometryCollection, please update the types to reflect that. :)  Thanks!

**Comment by deregtd** [comment]
FWIW, I've found that this happened only after using bbox to clip a polygon such that it was a degenerate polygon (two polygons with a 0-area connecting line/region).  I've stopped using bbox and instead use difference, which returns two non-degenerate polygons, and since then I haven't had the geometrycollection issue.  But either way, it feels like difference should either fix the types (and I'm guessing other turf functions can potentially return a collection under the similar circumstances) or throw errors in some fashion if you are using degenerate polys.
